sycl/doc/extensions/C-CXX-StandardLibrary/DeviceLibExtensions.rst

```diff
@@ -229,3 +229,19 @@ Those __devicelib_* functions have the same argument type and return type as cor
 complex math functions from <complex.h>, please refer to ISO/IEC 14882:2011 for details. The
 "double __complex__" type is C99 complex type and it is an alias to "struct {double, double}"
 in LLVM IR and SPIR-V.
+
+cl_intel_devicelib_cstring
+==========================
+
+.. code:
+   void *__devicelib_memcpy(void *dest, const void *src, size_t n);
```

Comment:
does this work for any combination of address spaces pointed to by dest and src? Might be worth adding a note. Std C library is not concerned with address spaces.

Reply:
Hi, @kbobrovs 
I think no matter what combination of address spaces pointed to by dest and src is, it should work if we can "read" from src and "write" to "dest". Is this correct?
Thanks very much.

Reply:
I don't know. I would expect this to work with any combination. My point that such details should be added to the user API description.

Reply:
Actually it will only work for `__generic` address space pointers.  Is there a way in DPC++ to specify `__generic` pointer explicitly?

Reply:
By default, function pointer arguments are generic. But users needs to understand if he/she can do e.g.

int arr[100];
...
auto *s1 = local_accessor.get_pointer();
auto *s2 = &arr[0];
__devicelib_memcpy(s1, s2, 100);

Reply:
Hi, @kbobrovs and @vzakhari 
__devicelib_memcpy is called by memcpy whose input pointer parameter is standard C/C++ pointer with address space.
So, if developer calls memcpy in SYCL device, the pointer is a generic_space pointer. And no matter where the input generic pointer points to, developers should ensure that they have access to the memory.
I also created a PR for memory function test: https://github.com/intel/llvm-test-suite/pull/371 which includes some basic tests and alignment test, I will update it to add some tests covering different address space memory.
Thanks very much.